### ISSUE TITLE
[0.x] Web Workers return 404

### ISSUE THREAD

**Issue by DarkGhostHunter** [laravel/vite-plugin#112]
- Laravel Vite Plugin Version: 0.5.2
- Laravel Version: 9.22.1
- Node Version: 17.2.0
- NPM Version: 8.15.1
- Host operating system: Windows 10 / Ubuntu 22.04
- Web Browser & Version: Chrome 105.0.5195.10
- Running in Sail / Docker: No

### Description:

Using `monaco-editor`, which requires Web Workers to be registered to enable functionality, [using it as instructed](https://twitter.com/youyuxi/status/1355316139144970240), the Web Workers are requested with invalid URLs.

![image](https://user-images.githubusercontent.com/5141911/182252507-49a0db49-cc94-42b1-b385-c741f2575218.png)

![image](https://user-images.githubusercontent.com/5141911/182252515-6bc99f91-8e4b-45c8-a60f-3ebe2a18d5c1.png)

Note that the workers URL is generated incorrectly by Vite by appending `/` to the worker URL:

![image](https://user-images.githubusercontent.com/5141911/182255976-defb72fe-a83c-40ce-8c55-9e0d0e0b7bd3.png)

[Details in this Stack Overflow question](https://stackoverflow.com/questions/73199951/monaco-editor-web-workers-are-registered-wrongly-return-404).

### Steps To Reproduce:

    npm install monaco-editor

```vue
<script setup>
import {onMounted, onUnmounted, ref, defineEmits} from "vue"
import * as monaco from "monaco-editor";
import editorWorker from "monaco-editor/esm/vs/editor/editor.worker?worker";
import jsonWorker from "monaco-editor/esm/vs/language/json/json.worker?worker";
import cssWorker from "monaco-editor/esm/vs/language/css/css.worker?worker";
import htmlWorker from "monaco-editor/esm/vs/language/html/html.worker?worker";
import tsWorker from "monaco-editor/esm/vs/language/typescript/ts.worker?worker";

self.MonacoEnvironment = {
    getWorker(_, label) {
        switch (label) {
            case 'json':
                return new jsonWorker();
            case 'css':
            case 'scss':
            case 'less':
                return new cssWorker();
            case 'html':
            case 'handlebars':
            case 'razor':
                return new htmlWorker();
            case 'typescript':
            case 'javascript':
                return new tsWorker();
            case 'yaml':
                return new yamlWorker();
            default:
                return new editorWorker()
        }
    }
};

monaco.languages.typescript.typescriptDefaults.setEagerModelSync(true);

const props = defineProps({
    modelValue: {
        type: String,
        default: '{"foo":"bar"}'
    },
});

const emit = defineEmits(['update:modelValue']);

const container = ref()

let instance = null

onMounted(() => {
    instance = monaco.editor.create(container.value, {
        model: monaco.editor.createModel(props.modelValue, 'json'),
        tabSize: 2,
    })

    instance.onDidChangeModelContent(() => {
        emit('update:modelValue', instance.getValue());
    });
})

onUnmounted(() => {
    instance?.dispose()
})
</script>

<template>
    <div ref="container"></div>
</template>
```

**Comment by jessarcher** [comment]
Hi @DarkGhostHunter, this looks like the same issue as #106.

I believe this is an upstream issue with Vite, but I'd be curious to know whether the problem exists *without* this plugin, by manually configuring Vite following https://vitejs.dev/guide/backend-integration.html

**Comment by DarkGhostHunter** [comment]
It's somewhat the same. My goal was to include [`monaco-yaml`](https://monaco-yaml.js.org/) as I need a YAML Editor plus Schema verification.

I'll try the manual backend config.

**Comment by driesvints** [comment]
Please see the other issue, thanks.

**Comment by DarkGhostHunter** [comment]
> Hi @DarkGhostHunter, this looks like the same issue as #106.
> 
> I believe this is an upstream issue with Vite, but I'd be curious to know whether the problem exists _without_ this plugin, by manually configuring Vite following https://vitejs.dev/guide/backend-integration.html

I was able to replicate the problem, and it seems that when using a backend instead of using Vite as a proxy won't allow web workers because what you said earlier: the browser doesn't allow web worker to run from different ports, even in the same domain.

In other words, to this to successfully work, the only workaround is to access the server through Vite.

[I elevated this to Vite](https://github.com/vitejs/vite/issues/9523).
